File: screenshot-test.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage({ viewport: { width: 1400, height: 900 } });

  await page.goto('http://localhost:3000');
  await page.waitForTimeout(3000);

  // Click on first hot stock to show chart panel
  const hotStock = await page.$('.hot-stock-item');
  if (hotStock) await hotStock.click();

  await page.waitForTimeout(3000);

  await page.screenshot({ path: 'chart-intraday-test.png', fullPage: false });
  console.log('Screenshot saved to chart-intraday-test.png');

  await browser.close();
})();

File: screenshot_test.py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page(viewport={"width": 1400, "height": 900})

    page.goto("http://localhost:3000")
    page.wait_for_timeout(3000)

    # Click first hot stock to open chart panel
    hot_stock = page.query_selector(".hot-stock-item")
    if hot_stock:
        hot_stock.click()

    page.wait_for_timeout(3000)

    page.screenshot(path="chart-intraday-test.png")
    print("Screenshot saved to chart-intraday-test.png")

    browser.close()
